spec/plsql/procedure_spec.rb

```diff
@@ -56,83 +57,59 @@
 
   end
 
-  describe "Function with numeric parameters" do
+  ['VARCHAR','VARCHAR2'].each do |datatype|
+    describe "Function with #{datatype} parameters" do
+      it_should_behave_like "Function with string parameters", datatype
+    end
+  end
+
+  shared_examples "Function with numeric" do |ora_data_type, class_, num1, num2, expected, mandatory|
     before(:all) do
       plsql.connect! CONNECTION_PARAMS
       plsql.execute <<-SQL
         CREATE OR REPLACE FUNCTION test_sum
-          ( p_num1 NUMBER, p_num2 NUMBER )
-          RETURN NUMBER
+          ( p_num1 #{ora_data_type}, p_num2 #{ora_data_type} )
+          RETURN #{ora_data_type}
         IS
         BEGIN
           RETURN p_num1 + p_num2;
         END test_sum;
       SQL
-      plsql.execute <<-SQL
-        CREATE OR REPLACE FUNCTION test_number_1
-          ( p_num NUMBER )
-          RETURN VARCHAR2
-        IS
-        BEGIN
-          IF p_num = 1 THEN
-            RETURN 'Y';
-          ELSIF p_num = 0 THEN
-            RETURN 'N';
-          ELSIF p_num IS NULL THEN
-            RETURN NULL;
-          ELSE
-            RETURN 'UNKNOWN';
-          END IF;
-        END test_number_1;
-      SQL
-      plsql.execute <<-SQL
-        CREATE OR REPLACE PROCEDURE test_integers
-          ( p_pls_int PLS_INTEGER, p_bin_int BINARY_INTEGER, x_pls_int OUT PLS_INTEGER, x_bin_int OUT BINARY_INTEGER )
-        IS
-        BEGIN
-          x_pls_int := p_pls_int;
-          x_bin_int := p_bin_int;
-        END;
-      SQL
     end
   
     after(:all) do
       plsql.execute "DROP FUNCTION test_sum"
-      plsql.execute "DROP FUNCTION test_number_1"
-      plsql.execute "DROP PROCEDURE test_integers"
       plsql.logoff
     end
   
-    it "should process integer parameters" do
-      expect(plsql.test_sum(123,456)).to eq(579)
-    end
-
-    it "should process big integer parameters" do
-      expect(plsql.test_sum(123123123123,456456456456)).to eq(579579579579)
+    it "should get #{ora_data_type} variable type mapped to #{class_.to_s}" do
+      expect( plsql.test_sum(num1, num2) ).to be_a class_
```

Comment:
This is related to the issue I've raised.
#85 but it's not a bug it's a difference between implementation of come Classes in Ruby/JRuby.
I've changed the unit tests to go above 20 digit for INTEGER to prove it can become a Bignum on both Ruby and JRuby.
I guess that the bug #85 can be closed/rejected as it's a feature, not a bug.